packages/smooth_app/lib/pages/prices/generic_infinite_scroll.dart

```diff
@@ -0,0 +1,314 @@
+class InfiniteScrollController<T, P> {
+  InfiniteScrollController({
+    required this.fetchItems,
+    Iterable<T> initialItems = _kEmptyIterable,
+    this.initialPage = 1,
+    this.onError,
+  })  : _currentPage = initialPage,
+        _items = List<T>.from(initialItems),
+        initialItems = List<T>.from(initialItems);
+
+  static const Iterable<Never> _kEmptyIterable = <Never>[];
+
+  /// Returns a Future with the fetched items and a boolean indicating if more items can be loaded
+  final Future<(List<T>, bool)> Function(P parameters, int page) fetchItems;
+
+  /// Parameters for the fetch operation
+  P? parameters;
+
+  /// Initial page number
+  final int initialPage;
+
+  /// Initial items to populate the list
+  final List<T> initialItems;
+
+  /// Called when an error occurs during fetching
+  final Function(dynamic error)? onError;
+
+  /// Current items in the list
+  List<T> _items;
+  List<T> get items => _items;
+
+  /// Current page being fetched
+  int _currentPage;
+  int get currentPage => _currentPage;
+
+  /// Whether currently loading more items
+  bool _isLoading = false;
+  bool get isLoading => _isLoading;
+
+  /// Whether more items can be loaded
+  bool _hasMoreItems = true;
+  bool get hasMoreItems => _hasMoreItems;
+
+  /// Additional pagination information
+  int? totalItems;
+  int? totalPages;
+
+  /// Load more items
+  Future<void> loadMore(P parameters) async {
+    if (_isLoading || !_hasMoreItems) {
+      return;
+    }
+
+    this.parameters = parameters;
+    _isLoading = true;
+
+    try {
+      final (List<T> newItems, bool hasMore) =
+          await fetchItems(parameters, _currentPage + 1);
+
+      _items.addAll(newItems);
+      _currentPage++;
+      _hasMoreItems = hasMore;
+    } catch (e) {
+      onError?.call(e);
+    } finally {
+      _isLoading = false;
+    }
+  }
+
+  /// Reset the controller to its initial state with optional new initial items
+  void reset({List<T>? newInitialItems}) {
+    _currentPage = initialPage;
+    _items = newInitialItems != null
+        ? List<T>.from(newInitialItems)
+        : List<T>.from(initialItems);
+    _isLoading = false;
+    _hasMoreItems = true;
+  }
+
+  /// Update pagination metadata
+  void updatePaginationInfo({int? newTotalItems, int? newTotalPages}) {
+    totalItems = newTotalItems;
+    totalPages = newTotalPages;
+  }
+}
+
+/// A generic stateful widget for infinite scrolling lists.
+class InfiniteScrollList<T, P> extends StatefulWidget {
+  const InfiniteScrollList({
+    super.key,
+    required this.controller,
+    required this.itemBuilder,
+    required this.loadingBuilder,
+    required this.errorBuilder,
+    required this.emptyBuilder,
+    required this.parameters,
+    this.loadMoreTriggerOffset = 200.0,
+    this.loadingMoreBuilder,
+    this.headerBuilder,
+    this.footerBuilder,
+    this.physics,
+    this.padding,
+    this.shrinkWrap = false,
+    this.onRefresh,
```

Comment:
Please no "just in case" parameters or default values.
From what I read in the code, the values are ALWAYS the same for the following parameters:
```dart
      loadMoreTriggerOffset: 200.0,
      loadingBuilder: (BuildContext context) =>
          const Center(child: CircularProgressIndicator()),
      errorBuilder: (BuildContext context, dynamic error) =>
          Text(error.toString()),
      emptyBuilder: (BuildContext context) => const Text(appLocalizations.prices_no_results),
      loadingMoreBuilder: (BuildContext context) => const Padding(
        padding: EdgeInsets.symmetric(vertical: 16.0),
        child: Center(child: CircularProgressIndicator()),
      ),
      footerBuilder: (BuildContext context) =>
          const SizedBox(height: 2 * MINIMUM_TOUCH_SIZE),
```